pkg/reconciler/messaging/channel/channel.go

```diff
@@ -20,374 +20,160 @@ import (
+	logger.Debug("Reconciling Channel", zap.Any("channel", c))
+	c.Status.InitializeConditions()
+	c.Status.ObservedGeneration = c.Generation
 
-	channel.Status.InitializeConditions()
-	channel.Status.ObservedGeneration = channel.Generation
-
-	// If ServiceAccountName is provided, reconcile workload identity.
-	if channel.Spec.ServiceAccountName != "" {
-		if _, err := r.Identity.ReconcileWorkloadIdentity(ctx, channel.Spec.Project, channel); err != nil {
-			return pkgreconciler.NewEvent(corev1.EventTypeWarning, workloadIdentityFailed, "Failed to reconcile Channel workload identity: %s", err.Error())
-		}
+	bcs := celltenant.StatusableFromChannel(c)
+	if err := r.Reconciler.ReconcileGCPCellTenant(ctx, bcs); err != nil {
+		logging.FromContext(ctx).Error("Problem reconciling Channel", zap.Error(err))
+		return fmt.Errorf("failed to reconcile Channel: %w", err)
+		//TODO instead of returning on error, update the data plane configmap with
+		// whatever info is available. or put this in a defer?
 	}
 
-	// 1. Create the Topic.
-	topic, err := r.reconcileTopic(ctx, channel)
-	if err != nil {
-		channel.Status.MarkTopicFailed("TopicReconcileFailed", "Failed to reconcile Topic: %s", err.Error())
-		return pkgreconciler.NewEvent(corev1.EventTypeWarning, reconciledTopicFailedReason, "Reconcile Topic failed with: %s", err.Error())
-	}
-	channel.Status.PropagateTopicStatus(&topic.Status)
-	channel.Status.TopicID = topic.Spec.Topic
-
 	// 2. Sync all subscriptions.
 	//   a. create all subscriptions that are in spec and not in status.
 	//   b. delete all subscriptions that are in status but not in spec.
-	if err := r.syncSubscribers(ctx, channel); err != nil {
+	if err := r.syncSubscribers(ctx, c); err != nil {
 		return pkgreconciler.NewEvent(corev1.EventTypeWarning, reconciledSubscribersFailedReason, "Reconcile Subscribers failed with: %s", err.Error())
 	}
 
-	// 3. Sync all subscriptions statuses.
-	if err := r.syncSubscribersStatus(ctx, channel); err != nil {
-		return pkgreconciler.NewEvent(corev1.EventTypeWarning, reconciledSubscribersStatusFailedReason, "Reconcile Subscribers Status failed with: %s", err.Error())
+	return pkgreconciler.NewEvent(corev1.EventTypeNormal, channelReconciled, "Channel reconciled: \"%s/%s\"", c.Namespace, c.Name)
+}
+
+func (r *Reconciler) FinalizeKind(ctx context.Context, c *v1beta1.Channel) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+	logger.Debug("Finalizing Channel", zap.Any("channel", c))
+	bcs := celltenant.StatusableFromChannel(c)
+	if err := r.Reconciler.FinalizeGCPCellTenant(ctx, bcs); err != nil {
+		return err
+	}
+
+	err := r.deleteAllSubscriberTopicsAndPullSubscriptions(ctx, c)
+	if err != nil {
+		return err
 	}
 
-	return pkgreconciler.NewEvent(corev1.EventTypeNormal, reconciledSuccessReason, `Channel reconciled: "%s/%s"`, channel.Namespace, channel.Name)
+	return pkgreconciler.NewEvent(corev1.EventTypeNormal, channelFinalized, "Channel finalized: \"%s/%s\"", c.Namespace, c.Name)
```

Comment:
I think this might be a misleading diff.

The old line still exists (`reconciledSuccessReason` was replace by `channelReconciled`) on the new file's line 78.

```golang
	return pkgreconciler.NewEvent(corev1.EventTypeNormal, channelReconciled, "Channel reconciled: \"%s/%s\"", c.Namespace, c.Name)
```


This line correspond's to the old file's line 392:
https://github.com/google/knative-gcp/blob/c1fd5094c5d21fab13cd5dc42d3ccf0d2d4233bc/pkg/reconciler/messaging/channel/channel.go#L392

Reply:
I see, but can we remove the escape here?

Reply:
Oh, I interpreted your original comment incorrectly.

Done. Both here and in ReconcileKind.